File: Sorting/bubble_sort/bubble_sort_v2.js



function bubbleSort(arr){

    //const swap = (arr, idx1, idx2) => {
        //[arr[idx1], arr[idx2]] = [arr[idx2], arr[idx1]]
    //}

    for(let i = 0; i < arr.length - 1; i++){
        for(let j = 0; j < arr.length - 1; j++){
            console.log(arr, arr[j], arr[j + 1])
            if(arr[j] > arr[j + 1]){
                [arr[j], arr[j + 1]] = [arr[j + 1], arr[j]]
            }
        }
    }
    return arr
}


console.log(bubbleSort([12, 1, 7, 5]))

File: Sorting/bubble_sort/bubble_sort_v2.py



def bubbleSort(arr):
    for i in range(len(arr) - 1):
        for j in range(len(arr) - 1):
            print(arr, arr[j], arr[j + 1])
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]

    return arr


bubbleSort([12, 1, 7, 5])
